src/main/java/co/cask/PluginJSON.java

```diff
@@ -196,6 +216,63 @@ public void execute()  throws MojoExecutionException {
   }
 
   /**
+   * Inspects all the Icon files to generate properties to be a part of the related widget.
+   */
+  private JSONObject getIcon(List<String> fileNames, File[] files, File nameFile)
+    throws MojoExecutionException {
+
+    JSONObject object = new JSONObject();
+    String name = FileUtils.removeExtension(nameFile.getName());
+    if (fileNames.contains(name)) {
+      object.put("type", "inline");
+      Map<String, String> arguments = new TreeMap<String, String>();
+      File file = files[fileNames.indexOf(name)];
+      try {
+        arguments.put("data", getDataURISchemaAsString(file, "data", "image/" + FileUtils.getExtension(
+          file.getName()), true));
+      } catch (Exception e) {
+        throw new MojoExecutionException(
+          String.format("Issue converting icon file '%s' to Base64 format. %s", file.getName(), e.getMessage())
+        );
+      }
+      object.put("arguments", arguments);
+    }
+
+    return object;
+  }
+
+  /**
+   * Returns the Data URI Scheme of File
+   */
+  private String getDataURISchemaAsString(File file, String scheme, String mediaType, boolean base64Extension)
+    throws URISyntaxException, IOException {
+    String path = scheme + ":" + mediaType;
+    path = base64Extension ? path.concat(";base64,") : path;
+
+    FileInputStream dataInFile = new FileInputStream(file);
+    byte[] dataInFileBytes = new byte[(int) file.length()];
+    dataInFile.read(dataInFileBytes);
```

Comment:
this is not the correct way to read from a stream. It will read up to the byte array length, but potentially less. It looks like commons-codec has a Base64OutputStream that you can use in conjunction with a ByteArrayOutputStream to do this in a streaming fashion.